pinot-core/src/main/java/org/apache/pinot/core/transport/AsyncQueryResponse.java

```diff
@@ -39,10 +44,13 @@
 @ThreadSafe
 public class AsyncQueryResponse implements QueryResponse {
   private final QueryRouter _queryRouter;
-  private final long _requestId;
+
+  // TODO(egalpin): Remove the concept of truncated request ID in next major release after all servers are expected
+  //  to send back tableName in DataTable metadata
+  private final long _canonicalRequestId;
```

Comment:
Is the overall plan to use request id + table name as the unique identifier of the query? Can you share how do you plan to manage request id?
E.g. only have one request id on broker side, and server splits it into multiple requests? I'm also considering whether to enhance `InstanceRequest` to have a list of <query + searchSegments + optionalSegments>

Reply:
What is `query` in this case ? Is it the whole query string ?